simplesat/rules_generator.py

```diff
@@ -296,20 +310,42 @@ def _add_install_requires_rules(self, package, work_queue, requirements):
                 else None)
 
             if not dependency_candidates:
-                msg = ("No candidates found for requirement {0!r}, needed for "
-                       "dependency {1!r}")
-                raise NoPackageFound(
-                    pkg_requirement,
-                    msg.format(pkg_requirement.name, package),
+                pkg_msg = ("'{0.name} {0.version}' from"
+                           " '{0.repository_info.name}'")
+                pkg_str = pkg_msg.format(package)
+                req_str = str(pkg_requirement)
+                msg = ("Blocking package {0!s}: no candidates found for"
+                       " dependency {1!r}").format(pkg_str, req_str)
+                if self.strict:
+                    # We only raise an exception if this comes directly from a
+                    # job requirement. Unfortunately, we don't track that
+                    # explicitly because we push all of the work through a
+                    # queue. As a proxy, we can examine the associated
+                    # requirements directly.
+                    if len(requirements) == 1:
+                        raise MissingInstallRequires(pkg_requirement, msg)
+                    else:
+                        logger.warning(msg)
+                else:
+                    logger.info(msg)
```

Comment:
[Nevermind](https://github.com/enthought/sat-solver/pull/156/files#diff-27965fdafe1fac1e643870b7f741db60R337), it's down below. We have to make our rule preventing installation first.